### PR TITLE
Skip all but exactly one return

### PR DESCRIPTION
**Pull request by TomasVotruba**
- add failing fixture
- skip all but exactly one return

### CODE REVIEW

File: rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictReturnExprRector.php [hunk 1]
```diff
@@ -80,12 +80,26 @@ public function refactor(Node $node): ?Node
...
+        return PhpVersion::PHP_74;
+    }
+
     private function hasSingleStrictReturn(ClassMethod $classMethod): bool
     {
         if ($classMethod->stmts === null) {
             return false;
         }
 
+        if ($this->betterNodeFinder->hasInstancesOf($classMethod->stmts, [Yield_::class])) {
```
Comment (samsonasik) [inline rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictReturnExprRector.php:94]: should exclude inner function, there is `hasInstancesOfInFunctionLikeScoped` method for it:

```suggestion
        if ($this->betterNodeFinder->hasInstancesOfInFunctionLikeScoped($classMethod, [Yield_::class])) {
```

File: rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictReturnExprRector.php [hunk 2]
```diff
@@ -80,12 +80,26 @@ public function refactor(Node $node): ?Node
...
+            return false;
+        }
+
+        $returns = $this->betterNodeFinder->findInstanceOf($classMethod->stmts, Return_::class);
```
Comment (samsonasik) [inline rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictReturnExprRector.php:98]: should exclude inner function, there is `findInstancesOfInFunctionLikeScoped` method for it:

```suggestion
        $returns = $this->betterNodeFinder->findInstancesOfInFunctionLikeScoped($classMethod, Return_::class);
```

Comment (TomasVotruba) [reply rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictReturnExprRector.php:98]: Thanks. Could you send those 2 changes in PR?

Comment (samsonasik) [reply rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictReturnExprRector.php:98]: I will try.

Comment (samsonasik) [reply rules/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictReturnExprRector.php:98]: https://github.com/rectorphp/rector-src/pull/2566

**Pull request merged**